src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -502,22 +505,40 @@ async def build_pex(
 def _validate_metadata(
     metadata: LockfileMetadata, request: PexRequest, python_setup: PythonSetup
 ) -> None:
-    if metadata.is_valid_for(
+    validation = metadata.is_valid_for(
         request.requirements.lockfile_hex_digest,
         request.interpreter_constraints,
         python_setup.interpreter_universe,
-    ):
-        return None
-
-    message = (
-        f"Invalid lockfile for PEX request `{request.output_filename}`."
-        "\n\n"
-        "If your requirements or your project's interpreter constraints have changed since the "
-        "lockfile was generated, you can follow the instructions in the header of the lockfile to "
-        "regenerate it. Otherwise, ensure your interpreter constraints setting is compatible with "
-        "the constraints specified in the lockfile."
     )
 
+    if validation:
+        return
+
+    message_parts = [
```

Comment:
There's a good chance that this code for generating the error message actually belongs in `lockfile_metadata.py`, with a more lightweight validation function remaining here